python/ray/_private/runtime_env/protocol.py

```diff
@@ -0,0 +1,96 @@
+import enum
+from ray._private.runtime_env.default_impl import get_protocols_provider
+
+
+class ProtocolsProvider:
+    @classmethod
+    def get_protocols(cls):
+        return {
+            # For packages dynamically uploaded and managed by the GCS.
+            "gcs",
+            # For conda environments installed locally on each node.
+            "conda",
+            # For pip environments installed locally on each node.
+            "pip",
+            # For uv environments install locally on each node.
+            "uv",
+            # Remote https path, assumes everything packed in one zip file.
+            "https",
+            # Remote s3 path, assumes everything packed in one zip file.
+            "s3",
+            # Remote google storage path, assumes everything packed in one zip file.
+            "gs",
+            # File storage path, assumes everything packed in one zip file.
+            "file",
+        }
+
+    @classmethod
+    def get_remote_protocols(cls):
+        return {"https", "s3", "gs", "file"}
+
+    @classmethod
+    def get_smart_open_transport_params(cls, protocol):
```

Comment:
This assumes a protocol must be supported by smart_open. Instead can we move download_and_unpack_package into a ProtocolProvider method that provides an impl for each protocol it provides?

Reply:
Updated.